### PR TITLE
[easy] Added `zip` and `pad` functionalities for arrays

### PR DESCRIPTION
**Pull request by querolita**
Introduced in https://github.com/o1-labs/o1js/pull/1848, creating a separate PR to facilitate the change earlier.

### CODE REVIEW

File: src/lib/util/arrays.ts [hunk 1]
```diff
@@ -12,3 +15,16 @@ function chunk<T>(array: T[], size: number): T[][] {
 function chunkString(str: string, size: number): string[] {
   return chunk([...str], size).map((c) => c.join(''));
 }
+
+function zip<T, S>(a: T[], b: S[]) {
+  assert(a.length === b.length, 'zip(): arrays of unequal length');
```
Comment (Geometer1729) [inline src/lib/util/arrays.ts:20]: AFAIK zip is usually implemented to take the minimum length when the lists are unequal. I think we should default to that unless there's a reason that wouldn't make sense here.

**Review by Geometer1729 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - Compiling stuck in the browser for recursive zkprograms https://github.com/o1-labs/o1js/pull/1906
 
+### Added
+
+- Provide `zip` and `pad` for the array utility https://github.com/o1-labs/o1js/pull/1921
```
Comment (Trivo25) [inline CHANGELOG.md:26]: nit: we actually dont expose the array utilities to the outside world directly, we use them as proxy most of the time - so these functions are just internal for now. I would remove the changelog entry but still merge the additions!

File: src/lib/util/arrays.ts [hunk 3]
```diff
@@ -12,3 +15,19 @@ function chunk<T>(array: T[], size: number): T[][] {
 function chunkString(str: string, size: number): string[] {
   return chunk([...str], size).map((c) => c.join(''));
 }
+
+function zip<T, S>(a: T[], b: S[]) {
+  assert(
+    a.length <= b.length,
+    'zip(): first array must be at least as long as the second array'
```
Comment (boray) [inline src/lib/util/arrays.ts:22]: message is not consistent with assertion.

Comment (querolita) [reply src/lib/util/arrays.ts:22]: 🤦🏻

**Pull request merged**